src/android/android-uvc.cpp

```diff
@@ -0,0 +1,598 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#ifdef RS2_USE_ANDROID_BACKEND
+
+
+#include "android-uvc.h"
+#include "../types.h"
+#include "libuvc/utlist.h"
+#include "usb_host/device_watcher.h"
+#include "../libuvc/utlist.h"
+
+
+#include <limits>
+
+
+#define MAX_PINS 5
+#define INTERRUPT_BUFFER_SIZE 0x400
+
+namespace librealuvc {
+    namespace platform {
+        // we are using standard fourcc codes to represent formats, while MF is using GUIDs
+        // occasionally there is a mismatch between the code and the guid data
+        const std::unordered_map<uint32_t, uint32_t> fourcc_map = {
+                {0x59382020, 0x47524559},    /* 'GREY' from 'Y8  ' */
+                {0x52573130, 0x70524141},    /* 'pRAA' from 'RW10'.*/
+                {0x32000000, 0x47524559},    /* 'GREY' from 'L8  ' */
+                {0x50000000, 0x5a313620},    /* 'Z16'  from 'D16 ' */
+                {0x52415738, 0x47524559},    /* 'GREY' from 'RAW8' */
+                {0x52573136, 0x42595232}     /* 'RW16' from 'BYR2' */
+        };
+
+        bool android_uvc_device::is_connected(const uvc_device_info &info) {
+            auto result = true;
+
+            //TODO: fix this
+            return result;
+        }
+
+        void android_uvc_device::init_xu(const extension_unit &xu) {
+            // not supported
```

Comment:
Perhaps we should throw a runtime error here?

Reply:
In the V4L backend the `init_xu` call seems to do nothing, but that's not the case in Windows.

For `set_xu` unsurprisingly both usbhost and V4L linux backend lead to `ioctl` calls, so I'm inclined to think `init_xu` should match too.

But this seems out of scope at the moment.